packages/x-telemetry/src/scripts/postinstall.ts

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable no-console */
+import fs from 'fs';
+import path from 'path';
+import { randomBytes } from 'crypto';
+import type { TelemetryContextType } from '../context';
+import getAnonymousEnvironment from '../internal/get-anonymous-environment';
+import getAnonymousProjectId from '../internal/get-project-id';
+import { TelemetryStorage } from '../internal/storage';
+import getMachineId from '../internal/get-machine-id';
+
+(async () => {
+  const storage = new TelemetryStorage({
+    distDir: path.join(process.cwd()),
+  });
+
+  const [anonymousEnvironment, projectId, machineId] = await Promise.all([
+    getAnonymousEnvironment(),
+    getAnonymousProjectId(),
+    getMachineId(),
+  ]);
+
+  const contextData: TelemetryContextType = {
+    config: {
+      isCollecting: storage.isCollecting,
+    },
+    traits: {
+      ...anonymousEnvironment,
+      machineId,
+      projectId,
+      sessionId: randomBytes(32).toString('hex'),
+      anonymousId: storage.anonymousId,
+    },
+  };
+
+  const writeContextData = (filePath: string, format: (content: string) => string) => {
+    const targetPath = path.resolve(__dirname, '..', filePath, 'context.js');
+    fs.writeFileSync(targetPath, format(JSON.stringify(contextData, null, 2)));
+  };
+
+  writeContextData('modern', (content) => `export default ${content};`);
```

Comment:
I was going to suggest storing it as a `.json` file, but TIL importing json file in ESM is not as easy as it was with `require` https://www.stefanjudis.com/snippets/how-to-import-json-files-in-es-modules-node-js/
Using a module makes sense 👍🏻